Three/include/CGAL/Three/Three.h

```diff
@@ -88,6 +89,19 @@ class THREE_EXPORT Three{
   static int default_normal_length;
   static int default_lines_width;
 
+public:
+  struct CursorScopeGuard
+  {
+    CursorScopeGuard(){}
+    void setCursor(QCursor cursor)
+    {
+      QApplication::setOverrideCursor(cursor);
+    }
```

Comment:
The real guard idiom would be to merge `setCursor` into the constructor.

Reply:
Which would trigger an unused variable warning. What they do in the example you cited is call release() in the end, but it's kind of the same, isn't it?

Reply:
Compiler are smart enough not to warn about that case. I think they warn about unused variables when they can prove that the constructor and the destructor have no side-effect.

[std::lock_guard](https://en.cppreference.com/w/cpp/thread/lock_guard) is an example, from the STL. Have a look at the example usage: only the constructor is called.